modules/memory-management-parent/memory-management/src/main/java/org/mule/runtime/internal/memory/management/DefaultMemoryManagementService.java

```diff
@@ -70,51 +77,54 @@ public void initialise() throws InitialisationException {
   @Override
   public synchronized ByteBufferProvider<ByteBuffer> getByteBufferProvider(String name, ByteBufferType byteBufferType,
                                                                            ByteBufferPoolConfiguration poolConfiguration) {
-    if (name == null) {
-      throw new IllegalArgumentException(BYTE_BUFFER_PROVIDER_NAME_CANNOT_BE_NULL_MESSAGE);
-    }
-
-    if (byteBufferProviders.containsKey(name)) {
-      throw new IllegalArgumentException(format(DUPLICATE_BYTE_BUFFER_PROVIDER_NAME, name));
-    }
-
-    byteBufferProviders.put(name, buildByteBufferProviderFrom(byteBufferType)
-        .withPoolConfiguration(poolConfiguration)
-        .withProfilingService(profilingService)
-        .withName(name)
-        .build());
-
-    return byteBufferProviders.get(name);
+    return doGetByteBufferProvider(name, byteBufferType, poolConfiguration);
   }
 
   @Override
   public synchronized ByteBufferProvider<ByteBuffer> getByteBufferProvider(String name, ByteBufferType byteBufferType) {
+    return doGetByteBufferProvider(name, byteBufferType, null);
+  }
+
+  private ByteBufferProvider<ByteBuffer> doGetByteBufferProvider(String name, ByteBufferType byteBufferType,
```

Comment:
Maybe it's better to synchronize over byteBufferProvidersUsageCount instead of the individual doGetByteBufferProvider and disposeByteBufferProvider methods. That way you can still work under the hipothetical case where someone is calling the doGet while another one is disposing the last one.